README
CPAN-Tester-WWW-Reports
=======================

This distribution contains the files used to run and manage the CPAN Testers 
Reports website.

The CPAN Testers Reports website is built on top of the Labyrinth Website 
Management System. As such, the code to drive the website is contained within
the Labyrinth and associated plugin prerequisites. In order to define the 
website in terms of functionality, layout and style, the files within the 
'./vhost' directory of this distribution should be installed into your web
server's virtual host directory.

INSTALLATION

This distribution is in two parts, the first is the installer and the second
are the website files.

In the first instance the installer will install Labyrinth and associated 
plugins. These library modules form the engine of the website. These can be
installed as

  perl Makefile.PL
  make
  make test
  make install

In the second instance, download the latest tarball and copy the contents of
the 'vhost' directory into the directory you plan to use to provide access to
the Admin. Create an appropriate database in MySQL and load the contents of the
SQL file, ./cgi-bin/db/cpanadmin-base.sql. Note that this only provides the 
base tables and contents.

You then need to update the configuration file, ./cgi-bin/config/settings.ini,
with the necessary database connection settings, together with the correct
paths to your virtual directory. Then you'll need update the run script,
./cgi-bin/pages.cgi, to point your configuration file.

Finally, you'll need to update your web server as necessary. The CPAN Testers 
Reports Administration website runs under Apache2, and the vhost file, 
./vhosts.conf, can be adapted as appropriate for other web servers 
applications.

CPAN TESTERS FUND

CPAN Testers wouldn't exist without the help and support of the Perl 
community. However, since 2008 CPAN Testers has grown far beyond the 
expectations of it's original creators. As a consequence it now requires
considerable funding to help support the infrastructure.

In early 2012 the Enlightened Perl Organisation very kindly set-up a
CPAN Testers Fund within their donatation structure, to help the project
cover the costs of servers and services.

If you would like to donate to the CPAN Testers Fund, please follow the link
below to the Enlightened Perl Organisation's donation site.

https://members.enlightenedperl.org/drupal/donate-cpan-testers

If your company would like to support us, you can donate financially via the
fund link above, or if you have servers or services that we might use, please
send an email to admin@cpantesters.org with details.

Our full list of current sponsors can be found at our I <3 CPAN Testers site.

http://iheart.cpantesters.org

SEE ALSO

* http://labyrinth.missbarbell.co.uk - Labyrinth

* http://www.cpantesters.org - CPAN Testers Reports website

AUTHOR

Barbie, <barbie@missbarbell.co.uk> for
Miss Barbell Productions, L<http://www.missbarbell.co.uk/>

COPYRIGHT AND LICENCE

Copyright (C) 2008-2014 Barbie for Miss Barbell Productions.

This distribution is free software; you can redistribute it and/or modify it 
under the Artistic License v2.

ADDITIONAL COPYRIGHT

This distribution includes TonyMCE [1], a Javascript WYSIWYG Editor used within
the site. TinyMCE is distributed under the GNU Lesser General Public License
Version 2.1, and all files within the './vhost/html/js/tiny_mce' directory
are copyright of the TinyMCE project.

[1] http://tinymce.moxiecode.com/



Changes
Changes for the CPAN Testers Reports website
============================================

        - stats code no longer to be stored as JSON, retrieved dynamically on
          request
        - new CGI script; reports-statistics.cgi, new & updated templates, and
          updated Javascript files.
        - perls & OSes no longer stored as JSON, retrieved dynamically on page
          request.
        - DB server now in US, so oldest request time are recorded at EDT.
        - Static site has now been deprecated. See GH#6 for more details:
          https://github.com/barbie/cpan-testers-www-reports/issues/6

3.59    2017-06-07
        - introduced auto incrementing storeid to stats_store, to make updates
          easier than lots of inserts.
        - updated scripts & files with changes from production environment.

3.58    2017-06-04
        - updated Release code to use new distribution indices.
        - upgrade to Labyrinth-5.32.
        - upgrade to Labyrinth-Plugin-Core-5.19.
        - upgrade to Labyrinth-Plugin-CPAN-3.54.
        - upgrade to Labyrinth-Plugin-Requests-1.08.
        - updated view-report.cgi to latest live version
          (thanks to Alceu Rodrigues de Freitas Junior for the patch)
        - updated changes from live production.
        - added Try::Tiny capture around the RemoveAuthorPages block due to
          some odd JSON parsing that was causing it to crash.
        - new code to load Perl/OS stats, due to the previous db overload.

3.57    2015-08-16
        - fixes from live.
        - updated quotation.
        - added debug lines to view-report.cgi.

3.56    2015-02-11
        - added Admin site links.
        - removed Perl promotions links.

3.55    2014-12-03
        - upgrade to Labyrinth-5.27.
        - upgrade to Labyrinth-Plugin-Core-5.18.
        - fixed RemoveAuthorPages as $name was the dist not the author name.
        - added admin menu links.

3.54    2014-09-23
        - XHTML template fixes.

3.53    2014-09-20
        - added documentation to:
          - reports-summary.cgi, reports-metadata.ci and release-summary.cgi.
        - fixed reports-summary.cgi to correctly reference the format as csv
          instead of txt. Updated templates.
        - fixed regex patterns matching perl development versions.
        - added ability to process command line options:
          - reports-summary.cgi, reports-metadata.ci and release-summary.cgi.
        - added JSON support (reports-summary.cgi).
        - author responses now include version for csv, json and xml output 
          (reports-summary.cgi).
        - fixed content type & disposition in return headers.

3.52    2014-09-20
        - fixed license fields in META.json to be lists.
        - added homepage to META resources.
        - added documentation about the CPAN Testers Fund.

3.51    2014-08-16
        - fixed distribution name in metadata.

3.50    2014-07-17
        - missing prerequisite in META.

3.49    2014-07-08
        - add logic for tester preferred contact address.
        - abstracted out Labyrinth-Plugin-CPAN.

3.48    2014-03-31
        - server side code for CPAN::Testers::WWW::Reports::Query::Report.

3.47    2014-03-16
        - exclude NA reports from the nopass RSS feeds (as requested by RJBS).
        - consolidate pre-requisites.

3.46    2014-03-02
        - removed 'reports_' as a prefix to database tables.

3.45    2014-03-02
        - layout fixes to static CSS.
        - rel=nofollow updates to iheart.js.
        - exclude rss/yml/json from msnbot crawlers.
        - various minor fixes.
        - upgrade to Labyrinth-5.20.
        - upgrade to Labyrinth-Plugin-Core-5.14.

3.44    2013-12-15
        - Added functionality for 'rmdist' and 'rmauth' for removing reports.
        - added missing prereq in makefile.PL
        - included new scripts used in live.
        - included missing templates used in live.
        - updated scripts and templates used in live.
        - fixed RSS dates as highlighted by Ricardo Signes.
        - restricted RSS feeds to one year, suggested by David Wheeler.
        - hacks to support RSS v2.0.

3.43    2013-07-10
        - added compress-reports.pl to start compressing reports in the 
          metabase table.

3.42    2013-06-30
        - minor displayed version fixes.
        - added new buildlevel to ensure overdue popular distributions are built 
          sooner rather than later.
        - allow build queries to be author or distro centric, or combined.
        - report serialisation switched to Data::FlexSerializer to compress 
          storage.

3.41    2013-03-24
        - fixed prerequisites.
        - removed redundant code.
        - added LICENSE.
        - added examples directory.
        - updated iheart to use JSON.

3.40    2012-09-06
        - reports-summary fixes.
        - distro.js template fix.
        - addition of reports-monitor.sh script.

3.39    2012-09-04
        - added iheart CSS.
        - versions with TRIAL now designated dev releases. [RT#77681 - CJM]
        - added distro version summary header. [RT#74905 - BAREFOOT]
        - Transform fix in the Builder.

3.38    2012-09-01
        - added crawler detection.
        - prevent Author & Distro crawler requests being passed to Builder.
        - upgraded Labyrinth Core to version 5.13.
        - change file dates changed to meet W3CDTF standards.

3.37    2012-07-08
        - renumbered versions to reflect website versions.
        - upgraded Labyrinth Core to version 5.12.
        - upgraded Labyrinth Core Plugin to version 5.10.
        - added sidebar sponsorship and advert panel.
        - reworked Makefile.PL for clarity.
        - added README
        - updated cpan-config.ini

3.36    2012-02-05
        - First public release (this was originally released as v1.00)

        # This release, although doesn't match the code released for the
        # the live website, forms the basis of the official public release

Makefile.PL
#!/usr/bin/perl

use strict;
use warnings;
use 5.006;

use ExtUtils::MakeMaker;

WriteMakefile(
    AUTHOR              => 'Barbie <barbie@cpan.org>',
    NAME                => 'CPAN::Testers::WWW::Reports',
    VERSION_FROM        => 'lib/CPAN/Testers/WWW/Reports.pm',
    ABSTRACT            => 'The CPAN Testers Reports website',
    NO_META             => 1,
    PREREQ_PM           => {

        # Labyrinth Modules
        'Labyrinth'                         => '5.32',
        'Labyrinth::DIUtils::ImageMagick'   => '5.04',
        'Labyrinth::Plugin::Core'           => '5.19',
        'Labyrinth::Plugin::Requests'       => '1.08',
        'Labyrinth::Plugin::CPAN'           => '3.54',

        # Perl Core Modules
        'CGI'                               => 0,
        'Carp'                              => 0,
        'Cwd'                               => 0,
        'Data::Dumper'                      => 0,
        'File::Basename'                    => 0,
        'File::Path'                        => 0,
        'IO::File'                          => 0,

        # Other Prereqs
        'Class::Accessor::Fast'             => '0',
        'Clone'                             => '0',
        'CPAN::Testers::Common::Article'    => '0',
        'CPAN::Testers::Common::Utils'      => '0',
        'CPAN::Testers::Fact::LegacyReport' => '0',
        'CPAN::Testers::Fact::TestSummary'  => '0',
        'Data::FlexSerializer'              => '0',
        'Email::Address'                    => '0',
        'File::Slurp'                       => '0',
        'GD::Graph::lines'                  => '0',
        'GD::Graph::colour'                 => '0',
        'HTML::Entities'                    => '0',
        'JSON::XS'                          => '0',
        'Metabase::Resource'                => '0',
        'Sort::Versions'                    => '0',
        'Time::Local'                       => '0',
        'version'                           => '0',
        'WWW::Mechanize'                    => '0',
        'XML::RSS'                          => '0',
        'YAML::XS'                          => '0',

        # for testing only
        'Test::More'                        => '0',

        # for toolkit scripts
        'File::Find::Rule'                  => '0',
        'Getopt::Long'                      => '0'

    }
);
